packages/react-conformance/src/defaultTests.tsx

```diff
@@ -73,4 +75,89 @@ export const defaultTests: TestObject = {
+        const rootPath = componentPath.replace(/src.*/, '');
+        const indexFile = require(path.join(rootPath, 'src', 'index'));
+
+        expect(indexFile[displayName]).toBe(Component);
+      });
+    }
+  },
+
+  /** Ensures component has top level file in package/src/componentName */
+  'has-top-level-file': (componentInfo: ComponentDoc, testInfo: TestingOptions) => {
+    if (!testInfo.isInternal) {
+      it(`has corresponding top-level file 'package/src/Component'`, () => {
+        const { displayName, componentPath, Component } = testInfo;
+        const rootPath = componentPath.replace(/src.*/, '');
+        const topLevelFile = require(path.join(rootPath, 'src', displayName));
+
+        expect(topLevelFile.default ? topLevelFile.default : topLevelFile[displayName]).toBe(Component);
+      });
+    }
+  },
+
+  /** If the component is a subcomponent, ensure its parent has the subcomponent as static property */
+  'is-static-property': (componentInfo: ComponentDoc, testInfo: TestingOptions) => {
+    it(`is a static property of its parent`, () => {
+      const { componentPath, displayName, useDefaultExport } = testInfo;
+      const componentFolder = componentPath.replace(path.basename(componentPath) + path.extname(componentPath), '');
+      const dirName = path.basename(componentFolder).replace(path.extname(componentFolder), '');
+      const isParent = displayName === dirName;
+
+      if (!isParent) {
+        const subComponentName = displayName.replace(dirName, '');
+        const subComponentFile = require(path.join(componentFolder, subComponentName));
+        const SubComponent = useDefaultExport ? subComponentFile.default : subComponentFile[subComponentName];
+
+        expect(ReactIs.isValidElementType(SubComponent)).toBe(true);
+      }
+    });
+  },
+
+  /** Ensures aria attributes are kebab cased */
+  'kebab-aria-attributes': (componentInfo: ComponentDoc, testInfo: TestingOptions) => {
+    it(`Aria attributes are kebab-cased`, () => {
+      const { Component, customMount, requiredProps } = testInfo;
+      const mountedComponent = customMount
+        ? customMount(<Component {...requiredProps} />)
+        : mount(<Component {...requiredProps} />);
+      const props = mountedComponent.props();
+
+      if (props) {
+        for (const prop of Object.keys(props)) {
+          if (prop.toLowerCase().includes('aria')) {
+            expect(prop.split(/[A-Z][a-z]+/)?.length).toBeLessThanOrEqual(1);
+            expect(prop.split('-').length).toBeGreaterThan(1);
+          }
+        }
+      }
+    });
+  },
+
+  /** Ensures that components have consistent custom callback names i.e. on[Part][Event] */
+  'consistent-callback-names': (componentInfo: ComponentDoc, testInfo: TestingOptions) => {
+    it(`has consistent custom callback names`, () => {
+      const { Component, customMount, requiredProps } = testInfo;
+      const mountedComponent = customMount
```

Comment:
Same as above, this needs to be tested on the declared props (`componentInfo.props`) not the rendered ones.